src/sagas/repository.ts

```diff
@@ -545,6 +545,89 @@ export function* getDatasetTablePreview({ payload }: any): any {
   }
 }
 
+/**
+ * Jobs.
+ */
+
+export function* getJobs({ payload }: any): any {
+  const { limit, offset, sort, direction, searchString } = payload;
+  try {
+    const response = yield call(
+      authGet,
+      `/api/repository/v1/jobs?offset=${offset}&limit=${limit}&sort=${sort}&direction=${direction}&filter=${searchString}`,
+    );
+    yield put({
+      type: ActionTypes.GET_JOBS_SUCCESS,
+      jobs: { data: response },
+    });
+  } catch (err) {
+    showNotification(err);
+    yield put({
+      type: ActionTypes.GET_JOBS_FAILURE,
+    });
+  }
+}
+
+/**
+ * Get a single result
+ */
+export function* getJobResult({ payload }: any): any {
+  const { id } = payload;
+  try {
+    const response = yield call(authGet, `/api/repository/v1/jobs/${id}`);
+    const jobStatus = response.data.job_status;
+    if (jobStatus !== Status.RUNNING) {
+      let resultResponse;
+      try {
+        resultResponse = yield call(authGet, `/api/repository/v1/jobs/${id}/result`);
+        yield put({
+          type: ActionTypes.GET_JOB_RESULT_SUCCESS,
+          payload: {
+            data: {
+              id,
+              jobResult: {
+                resultType: 'success',
+                result: resultResponse.data,
+              },
+            },
+          },
+        });
+      } catch (err: any) {
+        if (err.response) {
+          yield put({
+            type: ActionTypes.GET_JOB_RESULT_FAILURE,
```

Comment:
In this case, I think you might want to consider mapping this to GET_JOB_RESULT_SUCCESS.  If I understand correctly, you're dealing with HTTP response codes that would have typically tripped our pattern.  The request was successfully accepted by the server and the job status was successfully returned.  It just happened to be returned in a non-2XX response.  If the response payload doesn't contain any of the expected "error" details, I might then emit an ActionTypes.GET_JOB_RESULT_FAILURE.

Reply:
oh gotcha, updatd this, i think